src/sync/syncManagerOnline.ts

```diff
@@ -90,8 +90,8 @@ export function syncManagerOnlineFactory(
         // start syncing splits and segments
         if (pollingManager) {
 
-          // If singleSync is enabled pushManager and pollingManager should not start
-          if (singleSync === true) {
+          // If synchronization is disabled pushManager and pollingManager should not start
+          if (syncEnabled === false) {
```

Comment:
I would replace `if (syncEnabled === false)` with `if (syncEnabled)`, and shift the order of then/else branches